pkg/airgap/airgapinfo.go

```diff
@@ -0,0 +1,81 @@
+package airgap
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/pkg/errors"
+	kotsv1beta1 "github.com/replicatedhq/kotskinds/apis/kots/v1beta1"
+	"sigs.k8s.io/yaml"
+)
+
+type AirgapMetadata struct {
+	AirgapInfo   *kotsv1beta1.Airgap
+	K0sImageSize int64
+}
+
+// AirgapMetadataFromReader extracts the airgap metadata from the airgap file and returns it
+func AirgapMetadataFromReader(reader io.Reader) (metadata *AirgapMetadata, err error) {
+	metadata = &AirgapMetadata{}
+	// decompress tarball
+	ungzip, err := gzip.NewReader(reader)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decompress airgap file: %w", err)
+	}
+	defer ungzip.Close()
+
+	// iterate through tarball
+	tarreader := tar.NewReader(ungzip)
+	var nextFile *tar.Header
```

Comment:
any reason this needs to be defined outside the for loop?

Reply:
No reason. I'll update it.